src/main/resources/testchipip/csrc/cospike_impl.cc

```diff
@@ -277,8 +302,8 @@ int cospike_cosim(long long int cycle,
     for (int i = 0; i < info->nharts; i++) {
       // Use our own reset vector
       sim->get_core(hartid)->get_state()->pc = RESET_VECTOR;
-      // Set MMU to support up to sv39, as our normal hw configs do
-      sim->get_core(hartid)->set_impl(IMPL_MMU_SV48, false);
+      // Set MMU to support up to sv48, as our normal hw configs do
+      sim->get_core(hartid)->set_impl(IMPL_MMU_SV48, true);
```

Comment:
While SV48 is supported, there seems to be a bug during running Linux programs that causes divergence. Seems to be an issue where the PTW returns a PPN that is invalid.

Reply:
The checkpoint can be in a state where these bits are disabled, but the architectural state still matters. So we have to enable it so we can restore the registers, before restoring the X/F/VS bits to what the checkpoint expects.